Added check to make sure there's more than one image size.

jhned opened the pull request:
Fix for issue #121: So I checked for more than one image size by checking the sizes array of the WordPress image metadata array. If it's empty, then `tevkori_extend_image_tag` and `tevkori_filter_attachment_image_attributes` return `$html` and `$attr`, respectively.

I tested this on a fresh VVV install with a 100x100 image and a 500x500 image and got the expected results.

For better efficiency, we could pass through the `$img_meta` array to `tevkori_get_srcset_array`, which both of those functions already use. It would be a small duplication of code, but not necessarily a duplication of effort, since both filters will never be running at the same time.

joemcgill commented:
Hi @jhned, thanks for working through this. I think you're on the right track, but I'd suggest a little different approach. 

Rather than looking for all sizes returned in the attachment metadata, we can place check for the size of the srcset array in the `tevkori_get_srcset_array()` function, and return false if the array length is less than 2. This has two benefits: a) You don't have to make additional calls to `wp_get_attachement_metadata()` and b) You avoid a situation where there are several sizes available but only one would be returned as a source candidate because of divergent aspect ratios.

We may also need to update the filters to only run if the `srcset` is not false.

jhned commented:
Hey @joemcgill, sounds good. I added the check in the `tevkori_get_srcset_array()` function, and returned false if the array length was less than 2. To make sure that the `data-sizes` attribute didn't get set by accident, I moved the `srcset` check to the beginning of `tevkori_extend_image_tag` and `tevkori_filter_attachment_image_attributes`, returning the unaltered HTML and attributes if the `$srcset` is an empty string.

joemcgill commented:
@jhned much better. Now, the check for the number of sources should happen on the array created _after_ we've eliminated any sizes which have different aspect ratios. In other words, down at line 195:

```
if ( empty( $arr ) ) {
    return false;
}
```

Probably just change it to something like:

```
if ( count( $arr ) < 2 ) {
    return false;
}
```

tevko commented:
Hey @jhned , have you been able to make any progress on this? I'd like to get it merged into dev in time for our next release. Please continue to reach out if you need any help!

jhned commented:
Hey @tevko, I made the update, but I'm not sure what to do about the Travis CI test just yet. I'll try to look into it this evening. I haven't run unit tests in the past, but I've had a brief introduction.

joemcgill commented:
Looks like we would need to modify `test_tevkori_get_srcset_array_thumb()`, which expects to only find one image in the `srcset` array whenever the image is using the default thumbnail crop.

jaspermdegroot commented on wp-tevko-responsive-images.php:
```diff
@@ -260,6 +261,14 @@ function tevkori_get_src_sizes( $id, $size = 'thumbnail' ) {
+	$srcset = tevkori_get_srcset_string( $id, $size );
+
+	// If the srcset is empty, return the HTML unaltered.
+	if ( '' == $srcset ) {
```
I think this should check if ```srcset``` is not ```false``` instead of an empty string.

jaspermdegroot commented on wp-tevko-responsive-images.php:
```diff
@@ -287,6 +293,16 @@ function tevkori_extend_image_tag( $html, $id, $caption, $title, $align, $url, $
+	if ( ! isset( $attr['srcset'] ) ) {
+		$srcset = tevkori_get_srcset( $attachment_id, $size );
+
+		if ( '' == $srcset ) {
```
Same as above... check if ```srcset``` isn't ```false```.

tevko commented:
Closed by https://github.com/ResponsiveImagesCG/wp-tevko-responsive-images/pull/154
